### PR TITLE
How I Start Nim

### PR DESCRIPTION
**Pull request by def-**
Here's what I have so far. I'm still going to ask a few main Nim-people to proofread. Also, I ran into a few problems:

- Syntax highlighting on Github works great, but not with howistart. I guess pandoc or highlighting-kate are missing syntax highlighting for Nim
- Use bash highlighting or not? May look a bit weird
- My images and examples links may be wrong

I'd also like to see the final result before it's released, and I'm hoping for feedback.

### CODE REVIEW

**Comment by tsloughter** [comment]
This is great!

Is there a unit testing or other framework for Nim you could mention or use?

And with a language like Nim, being so new, if there are any CI services like Travis, Drone, Circle, that support Nim it would be nice to mention.

For highlighting, does any JS highlighter, like highlight.js, support Nim? I had to go with that for the Elixir post so could do similar again.

File: nim/1/index.md [hunk 1]
```diff
@@ -0,0 +1,807 @@
...
+million times. Try to compile and run it again. Notice how long it takes to run
+now. Is Nim's performance that abysmal? Of course not, quite the opposite!
+We're just currently building the binary in full debug mode, adding checks for
+integer overflows, array out of bounds and much more, as well as not optimizing
+the binary at all. The `-d:release` switch allows us to switch into release
+mode, giving us full speed:
+
+```bash
+$ nim c hello
+$ ./hello
```
Comment (tsloughter) [inline nim/1/index.md]: Should this be `$ time ./hello`?

Comment (def-) [inline nim/1/index.md]: Thanks, fixed.

**Comment by def-** [comment]
Added quite a bit. I didn't want to do testing, because I felt that the article was already pretty long and I had to do some refactoring, but now it's done.

I don't think any CI service supports Nim so far.

For highlighting I've always liked the output of pygments. highlight.js should work as well.

**Comment by tsloughter** [comment]
Awesome. I'll see if the highlight.js works well.

And you don't want this published yet, right? Waiting till after a Nim release?

**Comment by def-** [comment]
I think it's fine to be released now.

**Comment by tsloughter** [comment]
Ok, then I'll probably publish Monday morning. Gives me tomorrow to put it up on a staging app and try out highlight.js.

Will merge this PR in now though.

**Pull request merged**